### DISCUSSION TITLE
Help diagnosing issue 22.0.1

### DISCUSSION THREAD

**Discussion by roamerz** [keycloak/keycloak#22138] [Q&A]
Hi hoping to get some help diagnosing an issue.

In Safari IOS I'm getting an intermittent "Safari can't open the page because the network connection was lost" using any form of MFA (TOTP or SMTP)

The remote debugger console says "Error] Failed to load resource: The network connection was lost. (authenticate, line 0)"

The URL (real domain removed) that accompanies that message is "https://auth.xyz1234.com/realms/xxxxxx/login-actions/authenticate?session_code=JUhEN8JHNLkTgAmVPcTMF4HJjJ5t0pMUWkUf26NO5cA&execution=5b23341b-b618-4fb7-8658-68920ceb271c&client_id=https%3A%2F%2Fix2.xyz1234.com&tab_id=xZhX6uVTNiU"

The client is SAML. Don't see anything that sticks out in the logs. There are errors but they seem to happen even when it works.

Any idea where I might start looking?

**Comment by vmuzikar** [comment]
@roamerz Would you be able to provide a reproducer?

**Reply by roamerz** [reply]
@vmuzikar yes it can be reproduced. It seems to happen about 10% of the time.

**Reply by vmuzikar** [reply]
Would it be possible to share that reproducer? :)

**Reply by roamerz** [reply]
@vmuzikar  Hi sorry I am a dummy. What is a reproducer and what is the best way to share that?

**Reply by roamerz** [reply]
@vmuzikar I am able to reproduce this in Safari on macOS as well. I tried to send a timeline recording (json) but I was unable to upload it.

**Reply by vmuzikar** [reply]
We'd need some steps that'd allow us to reproduce the behaviour you are seeing on our end. What's your configuration that makes this issue to manifest?

**Comment by roamerz** [comment]
@vmuzikar Here are the steps required and some log entries that I saw that may be of interest. Also when I have key cloak running with message level set to debug, when the app fails there is no output to the key cloak console screen - just the timer kicking off every x seconds.

Would it be helpful to have a copy of our config? I'd hesitate to put it here in public though.

Open Safari and navigate to application
Application is not logged in so it forwards to Keycloak
Input username / password and click sign in

This is where it intermittently fails with:

Safari "Can't Open The Page" afari can’t open the page “https://xxxxx/login-actions/authenticate?session_code=qx-mfm0TS8WlE5ClprUCuzpHot5Sx7-b08IaTDSxp00&amp;execution=7f483d62-e7a2-480b-8830-d714f6d9c54f&amp;client_id=https%3A%2F%2Fix2.crimeix.com&amp;tab_id=_JZmpxaiC30” because the server unexpectedly dropped the connection. This sometimes occurs when the server is busy. Wait for a few minutes, and then try again.

In our use case it now forwards to the 2nd factor input page where you input the token
Input token and hit submit
If token is correct it then redirects back to our SAML application

We are currently using an SMTP token but the same happens with the normal TOTP token shipped with Keycloak.

When this happens nothing comes up in the console debug

**Reply by vmuzikar** [reply]
Thank you for providing more details. Seems like it might be also an issue with one of your custom providers (`com.mesutpiskin.keycloak.auth.email.EmailAuthenticatorForm`) but hard to tell without more details about that and also your config.

Possibly it may ring a bell to someone from @keycloak/core.

**Comment by roamerz** [comment]
Update: I have been doing some testing and this is happening without an MFA method attached to an auth flow - just straight username / password (local to keycloak)

I'm doing a fresh install to see if it's a bug associated with all the version upgrades this installation has been through.

**Comment by roamerz** [comment]
@vmuzikar 
Installing a complete new OS and KC did not solve the issue. I traced it to the firewall where there was some weirdness with either the IPS or SSL inspection. I suspect it was a bug because I removed then readded both to the policy and the problem hasn't happened since. 

On a positive note starting fresh has made KC much snappier. I did move to a postgres database at the same time so maybe that had something to do with that as well.

Anyhow thanks for responding to this call for help.
